Azure.ResourceManager.Core/Adapters/ModelAdapters.cs

```diff
@@ -17,6 +17,18 @@ protected TrackedResource(ResourceIdentifier id, Location location, T data)
             Model = data;
         }
 
+        protected TrackedResource(string id, Location location, T data)
+        {
+            if(object.ReferenceEquals(id, null)){
+                Id = null;
+            }
+            else{
+                Id = id;
+            }
+            Location = location;
+            Model = data;
+        }
+
```

Comment:
Why not simplify this to Id = object.ReferenceEquals(id, null)  ? null : id;?

This fails interestingly enough. 

The CIL ouput differs in that it puts one thing on the allocation stack and the set it at the end, making the runtime have to think whatever will be set will be a string, causing the unwanted implicit this to be call.


![image](https://user-images.githubusercontent.com/53361486/104252746-9f6b5000-5427-11eb-8ec1-b077752ca32a.png)